File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        
        int n = nums.length;
        int majority = n/2;
        HashMap<Integer, Integer> hm = new HashMap<>();

        for(int i=0; i<n; i++) {
            hm.put(nums[i], hm.getOrDefault(nums[i], 0)+1);
        }

        for(int i=0; i<n; i++) {
            if(hm.get(nums[i]) > majority) {
                return nums[i];
            }
        }
        return 0;

    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        Ele = 0
        for i in range(len(nums)):
            if count == 0:
                Ele = nums[i]
                count = 1
            elif nums[i] == Ele:
                count += 1
            else:
                count -= 1
        return Ele
